theano/gpuarray/opt.py

```diff
@@ -264,19 +264,26 @@ def __init__(self, local_optimizers_all, local_optimizers_map):
         self.local_optimizers_all = local_optimizers_all
         self.local_optimizers_map = local_optimizers_map
         self.failure_callback = None
+        self.new_opts = []
 
     def add_requirements(self, fgraph):
         fgraph.attach_feature(toolbox.ReplaceValidate())
 
+    def get_local_optimizers(self):
+        for opt in self.local_optimizers_all:
+            yield opt
+        # if repeat is not a problem we can drop the set
+        s = set()
+        for lopt in itervalues(self.local_optimizers_map):
+            for opt in lopt:
+                if opt not in s:
+                    yield opt
+                    s.add(opt)
+
     def apply(self, fgraph):
-        change_tracker = ChangeTracker()
         mapping = {}
-        global_process_count = {}
         start_nb_nodes = len(fgraph.apply_nodes)
         max_nb_nodes = len(fgraph.apply_nodes)
-        loop_timing = []
-        loop_process_count = []
-        local_opt_timing = []
         io_toposort_timing = []
         nb_nodes = []
```

Comment:
nb_nodes isn't used too. It is not needed as it was information per pass and we do only 1 here.